bin/runner.go

```
@@ -55,40 +53,54 @@ func main() {
 		// derive the required field from the experiment & engine and set into experimentDetails struct
 		if err := experiment.SetValueFromChaosResources(&engineDetails, clients); err != nil {
 			log.Errorf("Unable to set values from Chaos Resources, error: %v", err)
-			recorder.ExperimentSkipped(experiment.Name, utils.ExperimentNotFoundErrorReason)
+			msg := "Experiment Job creation failed, skipping Chaos Experiment: " + experiment.Name
+			events.SetEngineEventAttributes(&eventsDetails, utils.ExperimentNotFoundErrorReason, msg, "Warning", &engineDetails)
+			events.GenerateEvents(&eventsDetails, clients)
 			continue
 		}
 
 		// derive the envs from the chaos experiment and override their values from chaosengine if any
 		if err := experiment.SetENV(engineDetails, clients); err != nil {
 			log.Errorf("Unable to patch ENV, error: %v", err)
-			recorder.ExperimentSkipped(experiment.Name, utils.ExperimentEnvParseErrorReason)
+			msg := "Experiment Job creation failed, skipping Chaos Experiment: " + experiment.Name
+			events.SetEngineEventAttributes(&eventsDetails, utils.ExperimentEnvParseErrorReason, msg, "Warning", &engineDetails)
+			events.GenerateEvents(&eventsDetails, clients)
 			continue
 		}
 
 		log.Infof("Preparing to run Chaos Experiment: %v", experiment.Name)
 
 		if err := experiment.PatchResources(engineDetails, clients); err != nil {
 			log.Errorf("Unable to patch Chaos Resources required for Chaos Experiment: %v, error: %v", experiment.Name, err)
-			recorder.ExperimentSkipped(experiment.Name, utils.ExperimentDependencyCheckReason)
+			msg := "Experiment Job creation failed, skipping Chaos Experiment: " + experiment.Name
+			events.SetEngineEventAttributes(&eventsDetails, utils.ExperimentDependencyCheckReason, msg, "Warning", &engineDetails)
+			events.GenerateEvents(&eventsDetails, clients)
```

Btw, we should not ignore the errors.